plugins/guests/linux/cap/mount_smb_shared_folder.rb

```diff
@@ -82,27 +63,14 @@ def self.mount_smb_shared_folder(machine, name, guestpath, options)
           ensure
             # Always remove credentials file after mounting attempts
             # have been completed
-            machine.communicate.sudo("rm /etc/smb_creds_#{name}")
+            if !machine.config.vm.allow_fstab_modification
+              machine.communicate.sudo("rm /etc/smb_creds_#{name}")
```

Comment:
In order for the folder to be mounted the creds file need to exist on the guest. I think it makes sense to only remove the creds if fstab modification is turned off in this case. In order to make this safer, could make this file only readable by root?

Reply:
When a synced folder is removed these creds should probably be cleaned up as well

Reply:
Yes, setting the file mode to `0600` would be good, and removal when the fstab modification is not allowed is :+1:

Reply:
it gets set here https://github.com/hashicorp/vagrant/blob/1d602283a716e5514d20786bb4def9c60eb6a904/plugins/guests/linux/cap/mount_smb_shared_folder.rb#L42 🎉